metals/src/main/scala/scala/meta/internal/metals/ReferenceProvider.scala

```diff
@@ -48,6 +50,7 @@ final class ReferenceProvider(
   val index: TrieMap[Path, IdentifierIndex.MaybeStaleIndexEntry] =
     TrieMap.empty
   val identifierIndex: IdentifierIndex = new IdentifierIndex
+  val pcReferencesLock = new AtomicReference(Lock.completed)
```

Comment:
Wouldn't it make sense to use BatchedFunction ?

Reply:
I wanted to, but the way `BatchedFunction` works, you can always send a request instead of N requests. Not what we want to do. We have a pc for each build target, so those could only be grouped by build target. Still, I don't think it's what we want, I think we just want to cancel the previous request. Though there is one more thing I can change to reuse the created pc more...

Reply:
We should probably only focus on the latest request, you're right,